pkg/image/validation/os.go

```diff
@@ -217,3 +219,35 @@ func validateUnattended(def *image.Definition) []FailedValidation {
 
 	return failures
 }
+
+func validateRawConfig(def *image.Definition) []FailedValidation {
+	var failures []FailedValidation
+	isValidSize := regexp.MustCompile(`^([1-9]\d+|[1-9])+[MGT]`).MatchString
+
+	if def.OperatingSystem.RawConfiguration.DiskSize == "" {
+		return nil
+	}
+
+	if def.Image.ImageType != image.TypeRAW {
+		msg := fmt.Sprintf("The 'rawConfiguration/diskSize' field can only be used when 'image type' is '%s'.", image.TypeRAW)
+		failures = append(failures, FailedValidation{
+			UserMessage: msg,
+		})
+	}
+
+	if def.OperatingSystem.IsoInstallation.InstallDevice != "" {
+		msg := "You cannot simultaneously configure rawConfiguration and isoInstallation, regardless of image type."
+		failures = append(failures, FailedValidation{
+			UserMessage: msg,
+		})
+	}
+
+	if !isValidSize(def.OperatingSystem.RawConfiguration.DiskSize) {
+		msg := fmt.Sprintf("the 'rawConfiguration/diskSize' field must require an integer followed by an 'M/G/T' suffix to indicate size when 'imageType' is '%s'.", image.TypeRAW)
```

Comment:
"must require" is redundant. Consider simplifying to "... field must be an integer..."

Also, it should be "a M/G/T suffix", not "an". Or even better to avoid this entirely "... followed by a suffix of either 'M', 'G', or 'T'".

(yes, this feels excessive to point out, but it's user-facing messages so I tend to be on the more compulsive side of things)